### DISCUSSION TITLE
Cron and HTTP and HTTP Request

### DISCUSSION THREAD

**Discussion by AntonioSun** [deta/cloud-docs#577] [Help] [answered]
Since we can combine both cron and HTTP triggers in the same deta micro. I'm wonder how to use cron to trigger / simulate a http visit to my deta micro server. 

My micro service takes a long time and would normally take me several attempts to make it through, so I build a caching mechanism and hope my cron can hit my such micro service to update the cache periodically. 

Is it possible? Else I might have to write my own http request parser to do it. thx

**Update**, 

let's get into details since it is possible. Giving the following app:

```
const { App } = require('deta');
const express = require('express');

const app = App(express());

const myHandler = async(req, res) => {
    res.send('Hello Deta, I am running with HTTP');
});

app.get('/', myHandler);

app.lib.cron(event => {
    return 'Hello Deta, I am a cron job';
});

module.exports = app;
```

How to call `myHandler` from cron, simulating a web get request of HTTP Get `/?id=1&name=abc`?

**Comment by lemonyte** [comment, marked as answer]
You are able to call your HTTP route handlers from your cron function. Or if you feel like dipping your toes in the dark side of the pool, you can "simulate" an HTTP request to your Micro by using `fetch()` (JS), `requests` (Python), or any other way of sending HTTP requests, and sending a GET request to the chosen route, using the `DETA_PATH` env var to get the subdomain.

**Reply by AntonioSun** [reply]
let's get into details since it is possible. Please take a look at my updated OP, and see how to call myHandler from cron? Working solution appreciate sine I'm not a JS/Node developer (that can translate idea into code). thx

**Reply by lemonyte** [reply]
**Update in response to updated question**
The easy way (untested): 
```js
await fetch(`https://${process.env.DETA_PATH}.deta.dev/?id=1&name=abc`)
```
The harder way:
Call your handler function in the way that Express calls it, by manually creating request and response objects and passing them. This may or may not work, I'm not familiar enough with Express to say for certain.

**Reply by AntonioSun** [reply]
I looked it up and it seems `fetch` first appeared as an experimental feature in Node v17, and people normally need to [upgrade Node.js version to 17.5+](https://nodejs.org/dist/v17.5.0/). which means, my deta micro, which has a runtime of nodejs14.x, might not support it, right?

I.e., seems that dipping my toes in the dark side and call my handler function in the way that Express calls it, by manually creating request and response objects and passing them might be the only option left.

Oh, wait, that's the native `fetch`. There is also a `node-fetch`. I have to read more...

**Comment by XFeiLing** [comment]
您的邮件已经收到，谢谢！
